### ISSUE TITLE
Brave News on NTP consume a lot of CPU

### ISSUE THREAD

**Issue by atuchin-m** [brave/brave-browser#42021]
### Description

A report from @aseren:
> On the MacOS Development build from the fresh master, NTP with News enabled eats more then 50% of CPU core time.

### Steps to reproduce

1.  launch a browser with a clean profile
2.  open NTP
3.  enable Brave News
4. open internal task manager (NTP with news should be opened and active)
5. check CPU the avg CPU usage from `Tab: New Tab`  for 10-20 sec 

### Actual result

More than 20% avg CPU usage
![Image](https://github.com/user-attachments/assets/1be2fd40-e33d-4aa5-bbe7-5743e68d5cbf)

### Expected result

About zero CPU usage (< 5%)

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

Brave | 1.71.114 Chromium: 130.0.6723.58 (Official Build) (arm64)
-- | --
Revision | 89c4031c685a4296315a6f421f46275f0b72dd55
OS | macOS Version 15.0.1 (Build 24A348)

### Channel information

- [x] release (stable)
- [x] beta
- [x] nightly

### Reproducibility

- [x] with Brave Shields disabled
- [x] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by atuchin-m** [comment]
The issue is likely to be related `to components/common/useRelativeTime.ts` introduced in https://github.com/brave/brave-core/pull/25447

`getRelativeTime(time)` returns negative values:
<img width="70%" src="https://github.com/user-attachments/assets/7fcd95ca-cbce-49c3-a21b-df9142da864d" />
<img width="70%" src="https://github.com/user-attachments/assets/c587419f-0a2f-4375-92a8-98a362804a72" />

**Comment by kjozwiak** [comment]
The above requires `1.71.123` or higher for `1.71.x` verification 👍

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on

```
Brave | 1.71.123 Chromium: 130.0.6723.116 (Official Build) (64-bit)
-- | --
Revision | 59bc221d7f19c99ab7d3fc00748c1ad47776ae9f
OS | Windows 10 Version 22H2 (Build 19045.5011)
```
- Verified the STR from the description
- ensured that the CPU/GPU never increased more than ~5% using `1.71.123 Chromium: 130.0.6723.116`

Example | Example | Example
----------|-----------|---------
![Image](https://github.com/user-attachments/assets/10f31cbf-768c-41a6-bc3a-222ac0182a0b) | ![Image](https://github.com/user-attachments/assets/cdd29a73-d031-44ce-a76e-1228523c9966) | ![Image](https://github.com/user-attachments/assets/aca2e873-19bc-446b-afea-7c5b01009139)

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave | 1.71.123 Chromium: 130.0.6723.116 (Official Build) (x86_64)
-- | --
Revision | 59bc221d7f19c99ab7d3fc00748c1ad47776ae9f
OS | macOS Version 14.7.1 (Build 23H222)
```

**Comment by btlechowski** [comment]
Verified with

Brave | 1.71.123 Chromium: 130.0.6723.116 (Official Build) (64-bit)
-- | --
Revision | 59bc221d7f19c99ab7d3fc00748c1ad47776ae9f
OS | Linux


Reproduced the issue in 1.71.121

![Image](https://github.com/user-attachments/assets/d490d68a-b6a5-475e-bda9-36c760305ca8)


Verified fixed in 1.71.123

![Image](https://github.com/user-attachments/assets/c0ab6166-bf26-4a3e-8f8f-b164d45defac)
